src/util/byteswap.c

```diff
@@ -49,3 +49,27 @@ int32_t swap_int32( int32_t val )
     val = ((val << 8) & 0xFF00FF00) | ((val >> 8) & 0xFF00FF );
     return (val << 16) | ((val >> 16) & 0xFFFF);
 }
+
+uint64_t swap_uint64(uint64_t val)
+{
+    return (val & 0x00000000000000FFUL) << 56 | (val & 0x000000000000FF00UL) << 40 |
+             (val & 0x0000000000FF0000UL) << 24 | (val & 0x00000000FF000000UL) << 8 |
+             (val & 0x000000FF00000000UL) >> 8 | (val & 0x0000FF0000000000UL) >> 24 |
+             (val & 0x00FF000000000000UL) >> 40 | (val & 0xFF00000000000000UL) >> 56;
+}
+
+uint32_t swap_uint_length(uint32_t val, size_t bit_length)
+{
+    if (bit_length > 8) {
```

Comment:
good point, will add 64

I thought all one-line if statements did not comply to the style guide, but happy to make that change. 

-Brent